scripts/pi-hole/js/queries.js

```diff
@@ -22,62 +22,74 @@ function refreshData() {
 
 function add(domain,list) {
     var token = $("#token").html();
-    var alInfo = $("#alInfo");
-    var alList = $("#alList");
-    var alDomain = $("#alDomain");
-    alDomain.html(domain);
-    var alSuccess = $("#alSuccess");
-    var alFailure = $("#alFailure");
-    var err = $("#err");
-
-    if(list === "white")
-    {
-        alList.html("Whitelist");
-    }
-    else
-    {
-        alList.html("Blacklist");
-    }
+    var alertModal = $("#alertModal");
+    var alProcessing = alertModal.find(".alProcessing");
+    var alSuccess = alertModal.find(".alSuccess");
+    var alFailure = alertModal.find(".alFailure");
+    var alNetworkErr = alertModal.find(".alFailure #alNetErr");
+    var alCustomErr = alertModal.find(".alFailure #alCustomErr");
+    var alList = "#alList";
+    var alDomain = "#alDomain";
 
-    alInfo.show();
-    alSuccess.hide();
-    alFailure.hide();
-    $.ajax({
-        url: "scripts/pi-hole/php/add.php",
-        method: "post",
-        data: {"domain":domain, "list":list, "token":token},
-        success: function(response) {
-            if (response.indexOf("not a valid argument") >= 0 || response.indexOf("is not a valid domain") >= 0)
-            {
-                alFailure.show();
-                err.html(response);
-                alFailure.delay(4000).fadeOut(2000, function() { alFailure.hide(); });
-            }
-            else
-            {
-                alSuccess.show();
-                alSuccess.delay(1000).fadeOut(2000, function() { alSuccess.hide(); });
-            }
-            alInfo.delay(1000).fadeOut(2000, function() {
-                alInfo.hide();
-                alList.html("");
-                alDomain.html("");
-            });
-        },
-        error: function(jqXHR, exception) {
-            alFailure.show();
-            err.html("");
-            alFailure.delay(1000).fadeOut(2000, function() {
-                alFailure.hide();
-            });
-            alInfo.delay(1000).fadeOut(2000, function() {
-                alInfo.hide();
-                alList.html("");
-                alDomain.html("");
-            });
+    // Run only if the Modal is closed (multiple running interlock)
+    if (alertModal.css("display") === "none") {
```

Comment:
You can invert this if statement to reduce nesting.

Reply:
I'm not sure how to do this. Do you have any ideas or suggestions?

Reply:
For example:
```
if(condition) {
    // a
}
// b
```
Becomes inverted if we change it to
```
if(!condition) {
    // b
}
// a
```

With this, you can immediately return if the modal isn't closed, and the rest of the function isn't in one large if statement.

Reply:
After I made some tests I came to the conclusion that the whole code should stay inside the `if` condition to not break the multiple running interlock.

This interlock is intended to prevent the attempt to add another domain (when `add`-function gets called again) while the modal is open (because the _Black_-/_Whitelist_ buttons are sill reacheble with _Tab_, _Return_ and _Space_ keys, even if the Modal is shown).

Reply:
How would it break interlock, if it still is logically the same thing? If the modal is open, exit the method (inverted way). The code would look like this:
```
if (alertModal.css("display") !== "none")
    return;

// open modal, etc
```

If it is called again while the modal is open, it will just return at the if statement. The result is the same as if it was not inverted, except that there is less nesting.

Reply:
I think i've got it now.
`return` is the thing I was not thinking of.
Sorry, i'm no js expert